balrogscript/src/balrogscript/submitter/cli.py

```diff
@@ -342,9 +463,36 @@ def run(self, platform, productName, appVersion, version, build_number, locale,
 
         data.update(self._get_update_data(productName, version, build_number, **updateKwargs))
 
-        api = SingleLocale(name=name, build_target=build_target, locale=locale, auth0_secrets=self.auth0_secrets, api_root=self.api_root)
-        current_data, data_version = api.get_data()
-        api.update_build(data_version=data_version, product=productName, hashFunction=hashFunction, buildData=json.dumps(data), schemaVersion=9)
+        if self.backend_version == 2:
+            log.info("Using backend version 2...")
+            # XXX Check for existing data_version for this locale
+            blob = {
+                "blob": {"platforms": {build_target: {"locales": {locale: data}}}},
+                # XXX old_data_versions here is currently required but shouldn't be
+                "old_data_versions": {"platforms": {build_target: {"locales": {}}}},
+            }
+            headers = {"Accept-Encoding": "application/json", "Accept": "application/json", "Content-Type": "application/json", "Referer": self.api_root}
+            requests_api = get_balrog_api(auth0_secrets=self.auth0_secrets)
+            url = self.api_root + "/v2/" + name
+            logging.debug("Balrog request to {url} via POST", url)
```

Comment:
This code appears to be nearly identical or identical as an above block; it should probably be refactored. Do you think it belongs in `balrogclient`? I'm open to that.

Reply:
This does probably belong in balrogclient, I updated my apiv2 code to include that logging/timing fodder, which was culled from balrogclient from v1 code anyway.